src/Stratis.Bitcoin/Configuration/TextFileConfiguration.cs

```diff
@@ -17,6 +19,26 @@ public ConfigurationException(string message) : base(message)
         }
     }
 
+    /// <summary>
+    /// Abstract class for used with types requiring arguments when calling the ConvertValues method.
+    /// </summary>
+    public abstract class ConvertValueArguments
+    {
+    }
+
+    /// <summary>
+    /// Class used with the IPEndPoint type when calling the ConvertValues method.
+    /// </summary>
+    public class ConvertIPEndPointArguments:ConvertValueArguments
```

Comment:
This is very confusing to me, not sure what this solves.
If it is to inject the port and then retrieves the value as en IPEndpoint (with a port), then I think this is not the right solution. The user can just get the address and specify the port himself.

Reply:
The idea is to access all node settings arguments according to their final types so that we do not first read them as strings and then have conversions all over the place in our code.

Reply:
I find the second bit a lot less intuitive than the first one.

Reply:
Ok. Close this PR?

Reply:
I think so. At the moment we don't really need this and the solution is not very straightforward.
If you're into IP endpoints work, can you please check NBitcoin.Utils.ParseIpEndpoint and see how it compares to the extension method you just created?